Added functionality and error handling

puzzlepeaches opened the pull request:
Hey there!

We use this tool all the time and I'm getting a little better at Python so I thought I would make some contributions. The following changes were made:

- Changed requirements.txt to install the latest version of exchangelib (install was failing for me using current)
- Added error handling to exchangelib login functionality within modules such as gal
- Added the ability to print just the email to stdout when using gal module
- Added the ability to write emails dumped with the gal module to a file

Some things to note:

- Error handling isn't all-inclusive and does not deal with situations where the Exchange server isn't reachable very well just yet

Hope to add more functionality in the future and appreciate your hard work. Please let me know if you would like me to make any additional changes or modifications before we add this to master. Thanks!

cablethief commented:
Hi! Thanks for the interest and sorry for the delay, I will check out your request now.

puzzlepeaches commented:
Hey there!

I went ahead and:

* Removed version specification entirely from requirements.txt
* Added some additional exception handling to always print errors if we aren't sure what is failing in the try/except blocks
* Added help text to new flags in GAL function

The change that I made to the GAL dump functionality specifically removes the default printing of things such as:

```
Mailbox(name='John Smith', email_address='jsmith@acme.com', routing_type='SMTP', mailbox_type='Mailbox')
```

Instead, the function opts by default to print just the email address associated with an item. Adding -f prints the entire line shown above. This methodology fits my use case but can totally be pulled based on your preference. Let me know your thoughts. 

I am going to work on adding some more logic here to allow the user to specify if they only want to see users associated with a mailbox, etc. While working on the tool further, are there any changes/feature additions you would like me to take a whack at?

cablethief commented:
Coolio, thanks so much :D
